### PR TITLE
Cleanup various AWS SDK Bugs

### PR DESCRIPTION
**Pull request by lox**
This adds support for `BUILDKITE_ARTIFACT_UPLOAD_DESTINATION` containing a path (e.g s3://bucket/path). 

It also adds some more debugging around where the region is detected from and clears up a typo with env var names.

### CODE REVIEW

File: agent/s3.go [hunk 1]
```diff
@@ -1,41 +1,51 @@
...
 }
 
 func (e *credentialsProvider) Retrieve() (creds credentials.Value, err error) {
 	e.retrieved = false
 
-	creds.AccessKeyID = os.Getenv("BUILDKITE_S3_ACCESS_KEY_ID")
-	if creds.AccessKeyID == "" {
-		creds.AccessKeyID = os.Getenv("BUILDKITE_S3_ACCESS_KEY")
+	if v := os.Getenv(envS3AccessKeyID); v != "" {
+		logger.Debug("Found s3 access key id from " + envS3AccessKeyID)
```
Comment (lox) [inline agent/s3.go]: Accidentally dropped support for these variants. Misread this code.

**Comment by stefanmb** [comment]
@lox This fixes the auth problems I've encountered with the Beta agent, thanks!

File: agent/s3.go (hunk 1 above)
Comment (keithpitt) [inline agent/s3.go]: My brain did struggle to understand those changes! I was like "something looks wrong here, but I can't figure it out" - was hard to understand in the first place :)

**Comment by lox** [comment]
Going to split this into a few PRs. Conflated too many things!

**Comment by lox** [comment]
We will get a release out today @stefanmb.

**Pull request closed without merging**